source/agora/consensus/data/Enrollment.d

```diff
@@ -31,6 +31,9 @@ public struct Enrollment
     /// X: random seed, The nth image of random value
     public Hash random_seed;
 
+    /// Preimage information being revealed in advance
+    public PreimageInfo preimage;
```

Comment:
The `PreImageInfo` shouldn't be part of the enrollment, as the enrollment is written on the chain. It should be completely independent. A node can of course send a `PreImageInfo` along its enrollment (we can adapt the API to make this parameter optional), but it should also be able to receive, cache, and propagate PreImageInfo independently.

Reply:
The API endpoint named sendPreimage is planned to be made in another PR. And as you suggested, I will make the `enrollValidator` API function have the second parameter for a pre-image.